To-do list/Controllers/TaskDetailsViewController.swift

```diff
@@ -54,40 +50,92 @@ class TaskDetailsViewController: UIViewController {
         if  titleTextField.text?.isEmpty == true {
             Alert.showBasic(
                 title: "Please enter title of task",
                 vc: self)
-        } else if descriptionTextField.text?.isEmpty == true {
+            return false
+        } else if contentTextField.text?.isEmpty == true {
             Alert.showBasic(
                 title: "Please enter description of task",
                 vc: self)
+            return false
         } else {
-            let newTask = Task(title: titleTextField.text ?? "",
-                               description: descriptionTextField.text ?? "",
-                               deadline: dateTextField.text?.getDate() ?? Date(),
-                               status: task.status,
-                               id: task.id)
-            delegate?.taskDetails(TaskDetailsViewController(), didCreateUpdate: newTask)
-            navigationController?.popViewController(animated: true)
+            return true
+        }
+    }
+    
+    @IBAction func changeStatusButton(_ sender: UIButton) {
+        task.status = task.status.nextState
+        updateStatusUI(statusButton, task.status)
+    }
+    
+    @IBAction func submitButton(_ sender: UIButton) {
+        
+        if validateDataInput() {
+            guard let title = titleTextField.text ,
+            let content = contentTextField.text ,
+            let deadline = dateTextField.text ,
+            let status = statusButton.titleLabel?.text else {return}
+            
+            if task.id == 0 {
+                taskApiHelper.addTask(title: title,
+                                      content: content,
+                                      deadline: deadline,
+                                      status: status) { result in
+                    var message = String()
+                    if let result = result {
+                        if result.status == "ok"  {
+                            message = "Task added successfully"
+                            self.navigationController?.popViewController(animated: true)
+                        } else {
+                            message = "Session expired"
+                        }
+                    } else {
+                        message = "Something went wrong. Please try again."
+                    }
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.2){
+                        Alert.showBasicWithTimer(title: message, vc: self)
+                    }
+                }
+            } else {
+                
+                taskApiHelper.editTask(id: "\(task.id)",
+                                         title: title,
+                                         content: content,
+                                         deadline: deadline,
+                                         status: status) { result
+                    
+                    in
+                    var message = String()
+                    if let result = result {
```

Comment:
Tried to put var message like global var, in this case all messages ask to be self. and plus while testing, all messages got mixed. Then I tried to do func out of this two lines, one is String, other is T, because it can be any of Models that I get. Maybe 
//    func shortVersion <T> (result: T, message: String) -> (T, String) {
//        let result = result
//        var message = message
//        return result as! (T, String)
//    }

But then when I call in the same places, it doesn't work with logic.Please advise.

Reply:
what Denys is saying, result handling of API call is exactly same for addTask and editTask. You need to move this logic to a separate func to eliminate copypasta.  `message = "Task added successfully"` can be passed to the function


```
var message = String()
                     if let result = result {
                         if result.status == "ok"  {
                             message = "Task added successfully"
                             self.navigationController?.popViewController(animated: true)
                         } else {
                             message = APIHelper.ErrorAPI.otherError.description
                         }
                     } else {
                         message = APIHelper.ErrorAPI.networkError.description
                     }
                     DispatchQueue.main.asyncAfter(deadline: .now() + 0.2){
                         Alert.showBasicWithTimer(title: message, vc: self)
                     }

```